SMLHelper/Interfaces/IConsoleCommandHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace SMLHelper.V2.Interfaces
+{
+    using Commands;
+    using System;
+
+    /// <summary>
+    /// A handler class for registering your custom console commands.
+    /// </summary>
+    public interface IConsoleCommandHandler
+    {
+        /// <summary>
+        /// Registers your custom console command.
+        /// </summary>
+        /// <remarks>
+        /// <para>Target method must be <see langword="static"/>.</para>
+        /// 
+        /// <para>The command can take parameters and will respect optional parameters as outlined in the method's signature.<br/>
+        /// Supported parameter types: <see cref="string"/>, <see cref="bool"/>, <see cref="int"/>, <see cref="float"/>,
+        /// <see cref="double"/>.</para>
+        /// 
+        /// <para>If the method has a return type, it will be printed to both the screen and the log.</para>
+        /// </remarks>
+        /// <param name="command">The case-insensitive command to register.</param>
+        /// <param name="declaringType">The declaring type that holds the method to call when the command is entered.</param>
+        /// <param name="methodName">The name of the method to call within the declaring type when the command is entered. 
+        /// Method must be <see langword="static"/>.</param>
+        /// <param name="parameters">The parameter types the method receives, for targeting overloads.</param>
+        /// <seealso cref="ConsoleCommandAttribute"/>
+        void RegisterConsoleCommand(string command, Type declaringType, string methodName, Type[] parameters = null);
```

Comment:
Could this be expanded to include a non-Reflection based Register method?
```CS
void RegisterConsoleCommand(string commandText, Action commandAction);

void RegisterConsoleCommand<T>(string commandText, Action<T> commandAction);

void RegisterConsoleCommand<T1, T2>(string commandText, Action<T1, T2> commandAction);

void RegisterConsoleCommand<T1, T2, T3>(string commandText, Action<T1, T2, T3> commandAction);
```

Reply:
Since this is all text input, let me amend my examples
```CS
void RegisterConsoleCommand(string commandText, Action commandAction);

void RegisterConsoleCommand(string commandText, Action<string> commandAction);

void RegisterConsoleCommand(string commandText, Action<string, string> commandAction);

void RegisterConsoleCommand(string commandText, Action<string, string, string> commandAction);
```
And let the modder handle converting to their desired type

Reply:
Your suggested approach has the advantage of gaining maybe 0.2~ of a second of execution time once the user hits the enter button, but has the disadvantage of losing this standardised help text:
![image](https://user-images.githubusercontent.com/45315526/92755336-a1db7180-f383-11ea-9310-e49c5f30a9cf.png)

And personally, if they're going to have to convert from strings to their own type, they can still do so with this method.